### PR TITLE
[dx] Introduce git hooks and scripts for ESLint, Prettier, installing dependencies

### PR DESCRIPTION
**Pull request by wangannie**
# ✨ New in this PR ✨
Continuing the series of DX (Developer eXperience) improvements:
- Introduced pre-commit hooks using (husky and lint-staged) to lint and prettify any staged changes. This ensures that you can't commit any linter errors (but adding `--no-verify` to the commit command overrides this).
- Introduced [yarnhook](https://github.com/frontsideair/yarnhook#readme), which triggers `npm install` upon detecting changes to `package-lock.json` in order to ensure dependencies are properly updated on the following triggers:
  - `post-checkout` -- upon `git checkout` or `git switch`
  - `post-merge` -- on `git pull` on a local repo
  - `post-rewrite` -- on `git commit --amend` and `git rebase`
- Added helpful scripts for ESLint and Prettier
  - `npm run lint:check` -- run ESLint to detect errors/warnings
  - `npm run lint:fix` -- run ESLint and fix any resolveble issues
  - `npm run prettier:check` -- run Prettier to detect errors/warnings
  - `npm run prettier:fix` -- run Prettier and fix any issues

## How can the reviewer test your code? 👩‍💻
**Pre-commit hooks**
1. Make a change in a file that would violate a linter/prettier rule (e.g. add an unused import, unused variable, reorder imports)
   - Note: Command K + S to save without formatting (otherwise it might override your change)
    image
    image

2. Try to commit the changes. The linter will try to resolve issues that are automatically fixable (e.g. import ordering) and report other warnings or any remaining errors that couldn't be automatically resolved.
    image

**Hooks to install dependencies**
1. Switch to a different branch (e.g. `main` or another feature branch)
2. Switch back to this branch (`git checkout -`)
4. You should see `npm install` triggered. e.g. moving from `allison/sign-up-screen-styling` results in:
    ```sh
    Annies-MacBook-Pro-3:vouchers4veggies anniewang$ git checkout -
    Switched to branch 'annie/precommit-hooks'
    Changes to lockfile found, running `npm install`
    
    up to date in 2s
    
    179 packages are looking for funding
      run `npm fund` for details
    ```

## Any bugs you encountered or still having trouble with? 🐛
n/a

## Resources 📔
https://github.com/calblueprint/dccentralkitchen/pull/23
https://github.com/frontsideair/yarnhook#readme
https://git-scm.com/docs/githooks
https://prabinpoudel.com.np/articles/run-eslint-on-git-commit-with-husky-and-lint-staged/
https://github.com/okonet/lint-staged

🥦 CC: @oahnh

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
Comment (oahnh) [inline .husky/pre-commit:2]: Ran `npm install` and made an edit in StartScreen to import an unused var: 

image

Committing this change yields:
image

Are there other steps I need to take to setup the pre-commit hooks?

File: .husky/post-checkout [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
Comment (oahnh) [inline .husky/post-checkout:2]: Checking out also does not run `npm install`

**Review by oahnh — COMMENTED** [review]
Not sure if there are additional steps to configure aside from `npm install`, but I'm not able to use the hooks

**Review by sauhardjain — APPROVED** [review]
Works for me now, followed the instructions [here](https://prabinpoudel.com.np/articles/run-eslint-on-git-commit-with-husky-and-lint-staged/).

Thanks for this Annie — should be insanely helpful!

**Pull request merged**